frameworks/non-keyed/kobold/src/lib.rs

```diff
@@ -0,0 +1,226 @@
+use js_sys::Math;
+use kobold::prelude::*;
+use wasm_bindgen::prelude::*;
+
+const ADJECTIVES_LEN: usize = 25;
+static ADJECTIVES: &[&str; ADJECTIVES_LEN] = &[
+    "pretty",
+    "large",
+    "big",
+    "small",
+    "tall",
+    "short",
+    "long",
+    "handsome",
+    "plain",
+    "quaint",
+    "clean",
+    "elegant",
+    "easy",
+    "angry",
+    "crazy",
+    "helpful",
+    "mushy",
+    "odd",
+    "unsightly",
+    "adorable",
+    "important",
+    "inexpensive",
+    "cheap",
+    "expensive",
+    "fancy",
+];
+
+const COLOURS_LEN: usize = 11;
+static COLOURS: &[&str; COLOURS_LEN] = &[
+    "red", "yellow", "blue", "green", "pink", "brown", "purple", "brown", "white", "black",
+    "orange",
+];
+
+const NOUNS_LEN: usize = 13;
+const NOUNS: [&str; NOUNS_LEN] = [
+    "table", "chair", "house", "bbq", "desk", "car", "pony", "cookie", "sandwich", "burger",
+    "pizza", "mouse", "keyboard",
+];
+
+fn random(max: usize) -> usize {
+    (Math::random() * 1000.0) as usize % max
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+struct RowData {
+    id: usize,
+    label: String,
+}
+
+struct State {
+    rows: Vec<RowData>,
+    selected: Option<usize>,
+    last: usize,
+}
+
+impl State {
+    fn new() -> Self {
+        State {
+            rows: Vec::new(),
+            selected: None,
+            last: 0,
+        }
+    }
+    fn append_rows(&mut self, count: usize) {
+        self.rows.reserve_exact(count);
+        let d = self.last + 1;
+        for i in 0..count {
+            let adjective = ADJECTIVES[random(ADJECTIVES_LEN)];
+            let colour = COLOURS[random(COLOURS_LEN)];
+            let noun = NOUNS[random(NOUNS_LEN)];
+            let mut label = String::with_capacity(adjective.len() + colour.len() + noun.len() + 2);
```

Comment:
```suggestion
            let mut label = String::new();
            label.reserve_exact(adjective.len() + colour.len() + noun.len() + 2);
```

This may seem counter-intuitive, but it helps to ensure that pushing to the label reallocates...